### PR TITLE
Add convenient constructor for  `EllipticalSurface`

### PR DESCRIPTION
**Pull request by SebastianRuffert**

### CODE REVIEW

File: src/ConstructiveSolidGeometry/SurfacePrimitives/EllipticalSurface.jl [hunk 1]
```diff
@@ -18,12 +18,44 @@ Surface primitive describing circular bases, e.g. the top or bottom base of a [`
...
+
+#Type conversion happens here
+function EllipticalSurface{T}(r, φ, origin, rotation) where {T}
+    _r = _csg_convert_args(T, r)
+    (_φ, _rotation) = _handle_phi(_csg_convert_args(T, φ), rotation)
+    EllipticalSurface{T,typeof(_r),typeof(_φ)}(_r, _φ, origin, _rotation)
+end
+
+#Type promotion happens here
+function EllipticalSurface(::Nothing,r::TR, φ::TP, origin::PT, rotation::ROT) where {TR, TP, PT, ROT}
```
Comment (fhagemann) [inline src/ConstructiveSolidGeometry/SurfacePrimitives/EllipticalSurface.jl:37]: I am not in favor of this.. Is this `(::Nothing, ...)` really necessary?

File: src/ConstructiveSolidGeometry/SurfacePrimitives/EllipticalSurface.jl [hunk 2]
```diff
@@ -18,12 +18,44 @@ Surface primitive describing circular bases, e.g. the top or bottom base of a [`
...
+    EllipticalSurface{T}(r, φ, origin, rotation)
+end
+
+function EllipticalSurface(;
+    # define default parameters as Int to not influence type promotion
+    r = 1, 
+    φ = nothing,
+    origin = zero(CartesianPoint{Int}), 
+    rotation = one(SMatrix{3, 3, Int, 9}))
+    EllipticalSurface(nothing,r, φ, origin, rotation)#First argument is ugly workaround to distinguish promotion function and struct
```
Comment (fhagemann) [inline src/ConstructiveSolidGeometry/SurfacePrimitives/EllipticalSurface.jl:49]: You also write this in a comment here that this is an ugly workaround.
I guess, the definition of the `struct` already comes with a constructor for `EllipticalSurface`, even when no parametric types are passed. You might wanna specify the arguments up to a point where Julia knows which method to use (if the variables are not yet type promoted).

File: src/ConstructiveSolidGeometry/VolumePrimitives/Cone.jl [hunk 3]
```diff
@@ -156,8 +156,8 @@ function surfaces(t::Cylinder{T}) where {T}
     bot_center_pt = _transform_into_global_coordinate_system(CartesianPoint{T}(zero(T), zero(T), -t.hZ), t) 
     top_center_pt = _transform_into_global_coordinate_system(CartesianPoint{T}(zero(T), zero(T), +t.hZ), t) 
     mantle = FullConeMantle{T,:inwards}((t.r, t.r), t.φ, t.hZ, t.origin, t.rotation)
-    e_bot = CircularArea{T}(r = t.r, φ = t.φ, origin = bot_center_pt, rotation = t.rotation)
-    e_top = CircularArea{T}(r = t.r, φ = t.φ, origin = top_center_pt, rotation = -t.rotation * RotZ{T}(π))
+    e_bot = EllipticalSurface{T}(r = t.r, φ = t.φ, origin = bot_center_pt, rotation = t.rotation)
```
Comment (fhagemann) [inline src/ConstructiveSolidGeometry/VolumePrimitives/Cone.jl:159]: Are we still using the aliases `CircularArea` etc. somewhere?
If not, we might think about deleting them (?) from [lines 66-70 in EllipticalSurface.jl](https://github.com/SebastianRuffert/SolidStateDetectors.jl/blob/b09c61e4cbb952a39e64aa02b793669ba4fa3dc2/src/ConstructiveSolidGeometry/SurfacePrimitives/EllipticalSurface.jl#L66)

**Review by fhagemann — CHANGES_REQUESTED** [review]

File: src/ConstructiveSolidGeometry/VolumePrimitives/Cone.jl (hunk 3 above)
Comment (SebastianRuffert) [reply src/ConstructiveSolidGeometry/VolumePrimitives/Cone.jl:159]: image

File: src/ConstructiveSolidGeometry/SurfacePrimitives/EllipticalSurface.jl [hunk 4]
```diff
@@ -18,12 +18,44 @@ Surface primitive describing circular bases, e.g. the top or bottom base of a [`
 * `origin::CartesianPoint{T}`: The position of the center of the `EllipticalSurface`.
 * `rotation::SMatrix{3,3,T,9}`: Matrix that describes a rotation of the `EllipticalSurface` around its `origin`.
 """
-@with_kw struct EllipticalSurface{T,TR,TP<:Union{Nothing,T}} <: AbstractPlanarSurfacePrimitive{T}
-    r::TR = 1
-    φ::TP = nothing
+struct EllipticalSurface{T<:AbstractFloat,TR<:Union{T,Tuple{T,T}},TP<:Union{Nothing,T}} <: AbstractPlanarSurfacePrimitive{T}
```
[the remark this answers is not in the document]
Comment (SebastianRuffert) [reply src/ConstructiveSolidGeometry/SurfacePrimitives/EllipticalSurface.jl:21]: I thought so too, but this change did it

**Pull request merged**